presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaSecurityConfig.java

```diff
@@ -0,0 +1,142 @@
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_LOCATION_DOC)
+    public void setSslKeystoreFile(String sslKeystoreFile)
+    {
+        this.sslKeystoreFile = sslKeystoreFile;
+    }
+
+    @FileExists
+    public String getSslKeystoreFile()
+    {
+        return sslKeystoreFile;
+    }
+
+    @Config("kafka.keystore-password")
+    @LegacyConfig("kafka.ssl.keystore.password")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslKeystorePassword(String sslKeystorePassword)
+    {
+        this.sslKeystorePassword = sslKeystorePassword;
+    }
+
+    public String getSslKeystorePassword()
+    {
+        return sslKeystorePassword;
+    }
+
+    @Config("kafka.key-password")
+    @LegacyConfig("kafka.ssl.key.password")
+    @ConfigDescription(SslConfigs.SSL_KEY_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslKeyPassword(String sslKeyPassword)
+    {
+        this.sslKeyPassword = sslKeyPassword;
+    }
+
+    public String getSslKeyPassword()
+    {
+        return sslKeyPassword;
+    }
+
+    @Config("kafka.truststore-file")
+    @LegacyConfig("kafka.ssl.truststore.location")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_LOCATION_DOC)
+    public void setSslTruststoreFile(String sslTruststoreFile)
+    {
+        this.sslTruststoreFile = sslTruststoreFile;
+    }
+
+    @FileExists
+    public String getSslTruststoreFile()
+    {
+        return sslTruststoreFile;
+    }
+
+    @Config("kafka.truststore-password")
+    @LegacyConfig("kafka.ssl.truststore.password")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslTruststorePassword(String sslTruststorePassword)
+    {
+        this.sslTruststorePassword = sslTruststorePassword;
+    }
+
+    public String getSslTruststorePassword()
+    {
+        return sslTruststorePassword;
+    }
+
+    @Config("kafka.endpoint-identification-algorithm")
+    @ConfigDescription(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DOC)
+    public void setSslEndpointIdentificationAlgorithm(String sslEndpointIdentificationAlgorithm)
+    {
+        this.sslEndpointIdentificationAlgorithm = KafkaEndpointIdentificationAlgorithm.valueOf(sslEndpointIdentificationAlgorithm.toUpperCase(Locale.ENGLISH));
+    }
+
+    public KafkaEndpointIdentificationAlgorithm getSslEndpointIdentificationAlgorithm()
+    {
+        return sslEndpointIdentificationAlgorithm;
+    }
+
+    public Properties getKafkaClientProperties()
+    {
+        props = new Properties();
+        addNotNull(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, sslKeystoreFile);
+        addNotNull(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, sslKeystorePassword);
+        addNotNull(SslConfigs.SSL_KEY_PASSWORD_CONFIG, sslKeyPassword);
+        addNotNull(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, sslTruststoreFile);
+        addNotNull(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, sslTruststorePassword);
```

Comment:
Does it make sense to set `SSL_TRUSTSTORE_PASSWORD_CONFIG` when `sslTruststoreFile ` is `null`?

I think we need to capture invalid connector confiuration before trying to connect to Kafka. To fail fast. Otherwise it would be very difficult for users to understand why Presto can't connect Kafka.

The simpliest for me would be to require always all properties to be non null, but I am not sure if there are some exceptions here. Maybe `SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG` could be optional.

Reply:
From a logical standpoint yes, but as mentioned in another review comment: `kafka-clients` allows to set all those properties even when using `PLAINTEXT` - it shows up in the used config for the `consumer` and `producer` clients, but of course has no effect (haven't checkt if the lib complains when setting a password without the corresponding files, but you could specify a truststore and keystore even when using `PLAINTEXT`.

Currently all those porperties I added are optional - only if something is set by the user, then i forward this property - otherwise I don't forward it at all (also not assuming any defaults (i.e. derived from `kafka-clients`)...

-- Please let me know what your opinion is on this? Should I add such checks?